Bottom: 7fce83871b5006ad1e2b96344d1f006d9981dbc5
Top:    a94f3b20f99c2ff4808bcf412051b98a5a0cbb6a
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-10-16 16:23:02 +0100

Hack around Circuit CAM / EDG brokeness reading drill files











---

diff --git a/src/hid/gerber/gerber.c b/src/hid/gerber/gerber.c
index 1f7946a..877ee87 100644
--- a/src/hid/gerber/gerber.c
+++ b/src/hid/gerber/gerber.c
@@ -53,10 +53,16 @@ static void gerber_fill_polygon (hidGC gc, int n_coords, int *x, int *y);
 #define gerberYOffset(pcb, y) ((long) (-(y)))
 
 /* These are for drills */
-#define gerberDrX(pcb, x) ((long) ((x)/10))
-#define gerberDrY(pcb, y) ((long) (((pcb)->MaxHeight - (y)))/10)
-#define gerberDrXOffset(pcb, x) ((long) ((x)/10))
-#define gerberDrYOffset(pcb, y) ((long) (-(y))/10)
+#ifdef EDG_ARE_BROKEN
+#  define EDG_MUL 100
+#else
+#  define EDG_MUL 1
+#endif
+
+#define gerberDrX(pcb, x) ((long) (EDG_MUL * (x)/10))
+#define gerberDrY(pcb, y) ((long) (EDG_MUL * ((pcb)->MaxHeight - (y))/10))
+#define gerberDrXOffset(pcb, x) ((long) (EDG_MUL * (x) / 10))
+#define gerberDrYOffset(pcb, y) ((long) (EDG_MUL * -(y) / 10))
 
 /*----------------------------------------------------------------------------*/
 /* Private data structures                                                    */
